### DISCUSSION TITLE
Missing documentation for b:coc_diagnostic_info.lnums.

### DISCUSSION THREAD

**Discussion by adriantrunzo** [neoclide/coc.nvim#3121] [Q&A] [answered]
Hi all,

[The documentation](https://github.com/neoclide/coc.nvim/blob/1a74bf3c57fec8442f837b3baad0d6fb75d1b97a/doc/coc.txt#L1573) for the `b:coc_diagnostic_info` variable doesn't include the `lnums` property. That property seems great for showing the line number of the first error or warning in the statusline, but I don't want to rely on it if it is not officially supported.

When I trigger a warning, for example, this is the output I see for `echo b:coc_diagnostic_info`:

```
{'information': 0, 'hint': 0, 'lnums': [0, 21, 0, 0], 'warning': 1, 'error': 0}  
```

And indeed the warning is on line 21. If I trigger an error, I see the following:

```
'information': 1, 'hint': 0, 'lnums': [12, 0, 12, 0], 'warning': 0, 'error': 1}
```

From experimentation it seems that the `lnums` positions are as follows:

```
0: error line number
1: warning line number
2: information line number
3: hint line number
```

I've tried searching the source code for the definition of `lnums` but I couldn't find it. Can someone please point out the relevant source code to me? I'd like to learn why I cannot find it.

Will the `lnums` property be added to the documentation? Or is it an internal property that should not be used?

Thanks for the help!

**Comment by fannheyward** [comment, marked as answer]
Hello, the `coc_diagnostic_info` is defined in https://github.com/neoclide/coc.nvim/blob/9be0d9e053b3097c0bb84ea470217e9671fe0156/src/diagnostic/buffer.ts#L187-L208 the `lnums` is line number of the diagnostic.

`lnums = [0, 0, 0, 0]` => `[error, warning, info, hint]`, your understanding is correct.

**Reply by adriantrunzo** [reply]
Hi,

Thanks for the response @fannheyward! I realize now that the `/src` directory is not on the release (default) branch, which is why I could not find this code. Much appreciated!

I'd be happy to open a PR with an update to the documentation, if that is helpful.
